modules/auxiliary/scanner/rsync/modules_list.rb

```diff
@@ -25,147 +23,92 @@ def initialize
     )
     register_options(
       [
-        Opt::RPORT(873)
+        Opt::RPORT(873),
+        OptBool.new('AUTH_CHECK', [true, 'Check authentication or not', false])
       ], self.class)
   end
 
-  def read_timeout
-    10
-  end
+  def rsync(dir)
+    connect
 
-  def rsync_requires_auth?(rmodule)
-    sock.puts("#{rmodule}\n")
-    res = sock.get_once
-    if res && (res =~ /^#{RSYNC_HEADER} AUTHREQD/)
-      true
-    else
-      false
-    end
-  end
+    version = sock.get_once # server_initialisation
+    return if version.blank?
 
-  def rsync_list
-    sock.puts("#list\n")
+    sock.get # server_motd
```

Comment:
this will hang forever on rsync servers without a motd.  the timeouts used previous are necessary.